### PR TITLE
Add Search capabilities to claims/support/schools

### PR DESCRIPTION
**Pull request by CatalinVoineag**
## Context

We need to allow our support users to search schools by their names or postcodes. This commit adds this functionality with pg search.

It also adds pagy, for pagination and a search form partial that can be reused in other places.

We won't be implementing filters in this PR.

## Changes proposed in this pull request

- New pg search scope
- Refactored the support/schools controller a bit
- search form partial
- pagy gem

## Guidance to review

- Sign in as Colin in claims
- Search for a school by its name or postcode
- Clear the search by using the small `X` icon or by clicking `Clear search`
- Search for a school that doesn't exist.

## Screenshots

Prototype;
https://itt-mentoring-beta-94341ca35a2c.herokuapp.com/support/organisations
CleanShot_2024-01-03_at_14 26 29

https://github.com/DFE-Digital/itt-mentor-services/assets/11318084/9d47c063-0560-43b5-ac04-957fd73f0c86

### CODE REVIEW

**Comment by CatalinVoineag** [comment]
[As a Support User, I should be able to see the complete list of School Organisations](https://trello.com/c/QD099XPb/67-as-a-support-user-i-should-be-able-to-see-the-complete-list-of-school-organisations)

**Review by JamieCleare2525 — APPROVED** [review]
Tested locally. Really nice job!

**Review by elceebee — COMMENTED** [review]
Beautiful! I'm glad you were able to use the organisation list stuff, and I am now going to steal your search box!

I've just got a few comments.

File: app/assets/stylesheets/search-form.scss [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+.search-form .govuk-form-group{
+  flex: 1;
+}
+
+.search-form button {
+  margin-top: auto;
+  top: -2px;
+  margin-bottom: 0;
+  margin-left: 10px;
```
[the remark this answers is not in the document]
Comment (CatalinVoineag) [reply app/assets/stylesheets/search-form.scss:13]: I can use that to define the margin-left but I don't think I can use it for `top: -2px`

File: app/components/organisation_list_item.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+class OrganisationListItem < ApplicationComponent
+  attr_reader :organisation, :organisation_type, :organisation_url
+
+  def initialize(
+    organisation:,
+    organisation_type:,
```
[the remark this answers is not in the document]
Comment (CatalinVoineag) [reply app/components/organisation_list_item.rb:6]: OK, I'll let you change it as you know better how it needs to work with providers as well

File: app/controllers/claims/support/schools_controller.rb [hunk 3]
```diff
@@ -1,6 +1,9 @@
 class Claims::Support::SchoolsController < Claims::Support::ApplicationController
+  include Pagy::Backend
```
[the remark this answers is not in the document]
Comment (CatalinVoineag) [reply app/controllers/claims/support/schools_controller.rb:2]: No, I forgot to remove it

**Review by elceebee — APPROVED** [review]
Fabulous, thanks!

**Pull request merged**